tutorials/io/mergeSelective.py

```diff
@@ -0,0 +1,118 @@
+# 1. kOnlyListed: via <tt>TFileMerger::PartialMerge(kOnlyListed)</tt>
+#    This will merge only the objects in the files having the names in the
+#    specified list. If a folder is specified, its whole content will be merged
+#
+# 2. kSkipListed: via <tt>TFileMerger::PartialMerge(kSkipListed)</tt>
+#    This will skip merging of specified objects. If a folder is specified, its
+#    whole content will be skipped. 
+# 
+# Important note: the kOnlyListed and kSkipListed flags have to be bitwise OR-ed
+# on top of the merging defaults: kAll | kIncremental (as in the example)
+#
+# The files to be merged have the following structure:
+# - hpx          (TH1F)
+# - hpxpy        (TH2F)
+# - hprof        (TProfile)
+# - ntuple       (TNtuple)
+# - folder       (TDirectory)
+#      - hpx1    (TH1F)
+#
+# The example first merges exclusively hprof and the content of "folder",
+# producing the file exclusive.root, then merges all content but skipping
+# hprof and the content of "folder". The result can be inspected in the
+# browser.
+#
+# \macro_code
+#
+# \author The Root Team
+
+import ROOT
+
+def mergeSelective(nfiles=5):
+   # Create the files to be merged
+   tutdir = ROOT.gROOT.GetTutorialDir()
+   ret = ROOT.gROOT.LoadMacro(f"{tutdir}/hsimple.C")
+   if ret != 0:
+      raise Exception("Failed to load hsimple.C")
+      
+   files_to_cleanup = []
+   for i in range(0, nfiles):
+      files_to_cleanup.append(CreateFile(f"tomerge{i:03d}.root"))
+
+   #------------------------------------
+   # Merge only the listed objects
+   #------------------------------------
+   # NOTE: when the TFileMerger is used in a `with` statement, it will automatically
+   # close its output file when going out of scope.
+   with ROOT.TFileMerger(False) as fm:
+      fm.OutputFile("exclusive.root")
+      fm.AddObjectNames("hprof folder")
+      for i in range(0, nfiles):
+         fm.AddFile(f"tomerge{i:03d}.root")
+
+      # Must add new merging flag on top of the default ones
+      default_mode = ROOT.TFileMerger.kAll | ROOT.TFileMerger.kIncremental
+      mode = default_mode | ROOT.TFileMerger.kOnlyListed
+      fm.PartialMerge(mode)
+      fm.Reset()
+
+      #------------------------------------
+      # Skip merging of the listed objects
+      #------------------------------------
+      fm.OutputFile("skipped.root")
+      fm.AddObjectNames("hprof folder")
+      for i in range(0, nfiles):
+         fm.AddFile(f"tomerge{i:03d}.root")
+
+      # Must add new merging flag on top of the default ones
+      mode = default_mode | ROOT.TFileMerger.kSkipListed
+      fm.PartialMerge(mode)
+
+   # Cleanup initial files
+   for file in files_to_cleanup:
+      ROOT.gSystem.Unlink(file.GetName())
+
+   # Open files to inspect in the browser
+   ROOT.TFile.Open("exclusive.root")
+   ROOT.TFile.Open("skipped.root")
+   b = ROOT.TBrowser()
+
+
+def CreateFile(fname):
+   example = ROOT.gROOT.ProcessLineFast("hsimple(1)")
+   if example == 0:
+      return None
```

Comment:
Returning `None` from this function means we are storing a `None` in the list of files created above, that sounds dangerous.